src/PathUtilities.cpp

```diff
@@ -45,32 +45,31 @@ namespace
     using namespace Mezzanine;
 
     /// @brief Gets how many directories deep a path is.
-    /// @param StrIt The beginning of the range to check the depth of.
-    /// @param End The end of the range to check the depth of.
+    /// @param Directory The directory to check the depth of.
+    /// @param DirSep The directory separator(s) for splitting and rebuilding the path.
     /// @param ExitIfNegative If true, the function to return immediately if the depth count becomes negative.
-    /// @param Separator The directory separator for splitting and rebuilding the path.
     /// @return Returns an Integer representing how many directories down (or up, if negative) the path goes.
-    Integer GetDirectoryDepth_Common(StringView::const_iterator StrIt, StringView::const_iterator End,
-                                     const Boole ExitIfNegative, const Char8 Separator)
+    Integer GetDirectoryDepth_Common(const StringView Directory, const StringView DirSep, const Boole ExitIfNegative)
     {
         Integer Depth = 0;
-        String DirStr;
+        String TempDir;
 
-        while( StrIt != End )
+        StringView::const_iterator StrIt = Directory.begin();
+        while( StrIt != Directory.end() )
         {
-            if( (*StrIt) == Separator ) {
-                if( DirStr == ".." ) {
+            if( DirSep.find_first_of(*StrIt) != StringView::npos ) {
+                if( TempDir == ".." ) {
                     Depth -= 1;
```

Comment:
-- and ++?

Reply:
I don't see how there is an impactful difference between the two, but sure.